asterius/src/Asterius/Backends/Binaryen.hs

```diff
@@ -514,7 +515,16 @@ marshalExpression e = case e of
     if  | Just x <- SM.lookup unresolvedSymbol ss_sym_map ->
           lift $ Binaryen.constInt64 m $ x + fromIntegral symbolOffset
         | Just x <- SM.lookup unresolvedSymbol func_sym_map ->
-          lift $ Binaryen.constInt64 m $ x + fromIntegral symbolOffset
+          let base =
+                GetGlobal
+                  { globalSymbol = "__asterius_table_base",
+                    valueType = I32
+                  }
+           in marshalExpression $
+                addInt64
+                  (extendUInt32 base) -- TODO: or extendSInt32? Not sure which has the correct semantics.
```

Comment:
* `extendUInt32` is correct here, since we never handle negative base values
* We can perform `extendUInt32` on the result of adding the offset onto base; the offset would need to be an `i32` value then.

Reply:
> * `extendUInt32` is correct here, since we never handle negative base values

Right :+1: 

> * We can perform `extendUInt32` on the result of adding the offset onto base; the offset would need to be an `i32` value then.

Hmmm, not sure what you mean by this. How would you write this expression instead?

Reply:
``extendUInt32 (base `addInt32` fromIntegral (x + symbolOffset))``

This would also mean that we need to represent the symbol map with `SymbolMap Int` rather than the previous `SymbolMap Int32`. But changing the symbol map type can go in later PRs I think

Reply:
I see. Well, currently the `x` is an `Int64`, using the upper 32 bits for the tag so it really is more than `Int32`.